File: source/com/barrybecker4/simulation/cave/model/kernal/AbstractKernel.java
package com.barrybecker4.simulation.cave.model.kernal;

import com.barrybecker4.simulation.cave.model.Cave;

/**
 * Looks only at immediate neighbors
 * @author Barry Becker
 */
public abstract class AbstractKernel implements Kernel {

    protected Cave cave;

    public AbstractKernel(Cave cave) {
         this.cave = cave;
    }

    boolean isOffEdge(int x, int y) {
        return (x < 0 || y < 0 || x >= cave.getWidth() || y >= cave.getLength());
    }

}

File: scala-source/com/barrybecker4/simulation/cave/model/kernal/AbstractKernel.scala
// Copyright by Barry G. Becker, 2000-2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.simulation.cave.model.kernal

import com.barrybecker4.simulation.cave.model.Cave


/**
  * Looks only at immediate neighbors
  * @author Barry Becker
  */
abstract class AbstractKernel(var cave: Cave) extends Kernel {
  private[kernal] def isOffEdge(x: Int, y: Int) = x < 0 || y < 0 || x >= cave.getWidth || y >= cave.getLength
}
